GVRf/Framework/framework/src/main/java/org/gearvrf/GVRSceneObject.java

```diff
@@ -699,11 +699,20 @@ public GVRSceneObject getParent() {
      * @param child
      *            {@link GVRSceneObject Object} to add as a child of this
      *            object.
+     * @return true if child was added, else false
      */
-    public void addChildObject(GVRSceneObject child) {
+    public boolean addChildObject(GVRSceneObject child) {
+        if (child.mParent == this) {
+            return false;
+        }
+        if (child.mParent != null) {
+            Log.e("gvrf", "Cannot add child - it has another parent");
```

Comment:
How can you tell why false was returned? There are two separate conditions that are quite different that can give you that result.

If the operation of adding a child that already has a different parent is not allowed/an error then we should throw an exception. It seems to me like an abnormal operation and the users should know clearly it is an error they need to fix.